app/Http/Controllers/AttendanceController.php

```diff
@@ -53,6 +54,16 @@ public function store(StoreAttendanceRequest $request): JsonResponse
         // Variables for comparison below
         $date = $request->input('created_at', date('Y-m-d'));
         $gtid = $request->input('gtid');
+
+        // if an access card number is provided without a GTID, try to find the matching GTID
+        if ($request->input('gtid') === null && $request->input('access_card_number') !== null) {
+            if (AccessCard::where('access_card_number', '=', $request->input('access_card_number'))->exists()) {
+                $card = AccessCard::where('access_card_number', '=', $request->input('access_card_number'))->sole();
```

Comment:
I think you could combine this all into one `find()` call.

Reply:
Nope https://github.com/laravel/framework/blob/11.x/src%2FIlluminate%2FDatabase%2FQuery%2FBuilder.php#L2900